src/boot/device-path-util.c

```diff
@@ -186,6 +186,23 @@ size_t device_path_size(const EFI_DEVICE_PATH *dp) {
         return (const uint8_t*) i - (const uint8_t*) dp + sizeof(EFI_DEVICE_PATH);
 }
 
+size_t device_path_size_with_limit(const EFI_DEVICE_PATH *dp, size_t limit) {
+        const EFI_DEVICE_PATH *i = ASSERT_PTR(dp);
+        size_t size = 0;
+
+        /* Never read past 'limit' bytes, useful when processing userspace vars */
```

Comment:
maybe mention what happens if the limit is hit, i.e. that we truncate the size as the last item that completely fits under the limit, which possibly is zero